### PR TITLE
Extract json, nkey and jwt utils to libraries

### PR DESCRIPTION
**Pull request by scottf**
nkey and jwt code were extracted from the library to their own to make the jwt code available without the client. The json code was extracted because the jwt code depends on it.

In all cases, no public api was changed. This was accomplished in 2 ways.

1. In the case of nkey and json (with one exception), the extracted code was kept in the exact same package, class names did not change, and api did not change.
2. In the case of jwt, since there was only a single class with static api, most of which were deprecated, that class ended up delegating to the extract jwt library, which removed deprecated methods.
3. In the case of the JsonUtils, it was completely deprecated and methods that were not previously deprecated were delegated to the new JsonWriteUtils or extracted (i.e. HeaderUtils)

If the developer uses gradle or maven, extracted dependencies will automatically be retrieved because that's how their dependency management works, so there will be no compilation issues. Binary compatibility is not guaranteed, which is generally not an issue since the Java best practice always rebuilds code before deployment, but we are bumping the minor version to 18 for visibility.

### CODE REVIEW

File: build.gradle [hunk 1]
```diff
@@ -36,7 +36,10 @@ repositories {
 }
 
 dependencies {
-    implementation 'net.i2p.crypto:eddsa:0.3.0'
```
Comment (scottf) [inline build.gradle:39]: crypt is needed by the nkeys library, so not directly needed here

File: src/main/java/io/nats/client/support/JsonUtils.java [hunk 2]
```diff
@@ -109,9 +107,9 @@ public static String endFormattedJson(StringBuilder sb) {
      * @param json raw json
      */
     public static void addRawJson(StringBuilder sb, String fname, String json) {
-        if (json != null && json.length() > 0) {
+        if (json != null && !json.isEmpty()) {
```
Comment (scottf) [inline src/main/java/io/nats/client/support/JsonUtils.java:110]: IntelliJ won't stop harassing me!

File: src/test/java/io/nats/client/support/JwtUtilsTests.java [hunk 3]
```diff
@@ -48,7 +48,7 @@ public void issueUserJWTSuccessMinimal() throws Exception {
 
         String jwt = issueUserJWT(SIGNING_KEY, ACCOUNT_ID, new String(USER_KEY.getPublicKey()), null, null, null, 1633043378, "audience");
         String claimBody = getClaimBody(jwt);
-        String cred = String.format(NATS_USER_JWT_FORMAT, jwt, new String(USER_KEY.getSeed()));
+        String cred = String.format(io.nats.jwt.JwtUtils.NATS_USER_JWT_FORMAT, jwt, new String(USER_KEY.getSeed()));
```
Comment (MauriceVanVeen) [inline src/test/java/io/nats/client/support/JwtUtilsTests.java:51]: Could instead have one import of `io.nats.jwt.JwtUtils.NATS_USER_JWT_FORMAT` on top. But just a small thing.

**Review by MauriceVanVeen — APPROVED** [review]

File: src/test/java/io/nats/client/support/JwtUtilsTests.java (hunk 3 above)
Comment (scottf) [reply src/test/java/io/nats/client/support/JwtUtilsTests.java]: This needs to complete test the deprecated version of this field, so I did remove this fully qualified part, but so the field tested is the one in the deprecated class. I checked all the imports too.

**Pull request merged**